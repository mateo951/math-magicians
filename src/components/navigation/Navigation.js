import React from 'react';
import { NavLink } from 'react-router-dom';
import style from './nav.module.css';

const Navigation = () => (
  <nav>
    <NavLink className={style.listitems} to="/" exact>
      <h1>Math Magicians</h1>
    </NavLink>
    <NavLink to="/" exact>
      <img src="https://img.icons8.com/external-bearicons-flat-bearicons/64/000000/external-magic-wand-graphic-design-bearicons-flat-bearicons.png" alt="logo" />
    </NavLink>
    <ul className={style.navlinks}>
      <NavLink to="/" className={style.listitems} exact>
        Home
      </NavLink>
      <NavLink to="/calculator" className={style.listitems} exact>
        Calculator
      </NavLink>
      <NavLink to="/quotes" className={style.listitems} exact>
        Quotes
      </NavLink>
    </ul>
  </nav>
);

export default Navigation;
