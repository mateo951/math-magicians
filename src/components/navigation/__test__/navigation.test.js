import React from 'react';
import ReactDOM from 'react-dom';
import { render, screen } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
import Navigation from '../Navigation';

let div = null;
beforeEach(() => {
  div = document.createElement('div');
  document.body.appendChild(div);
  render(<Router><Navigation /></Router>);
});

describe('Test rendering', () => {
  it('renders with out crashing', () => {
    ReactDOM.render(<Router><Navigation /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders NavLink correctly', () => {
    expect(screen.getByText('Calculator')).toMatchSnapshot();
    expect(screen.getByText('Quotes')).toMatchSnapshot();
  });

  it('renders text correctly', () => {
    const calculator = screen.getByText('Math Magicians');
    expect(calculator).toMatchSnapshot();
  });
});
