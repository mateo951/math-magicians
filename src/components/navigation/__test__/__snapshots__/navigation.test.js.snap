// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering renders NavLink correctly 1`] = `
<a
  class="listitems"
  href="#/calculator"
>
  Calculator
</a>
`;

exports[`Test rendering renders NavLink correctly 2`] = `
<a
  class="listitems"
  href="#/quotes"
>
  Quotes
</a>
`;

exports[`Test rendering renders text correctly 1`] = `
<h1>
  Math Magicians
</h1>
`;
